### DISCUSSION TITLE
[Feature Request] double-single-quote variable format option to handle SQL strings containing multi-value dashboard variables

### DISCUSSION THREAD

**Discussion by podolsky** [grafana/grafana#49741] [Dashboards, Panels & Visualizations]
I had recent occasion to need to include a PostgreSQL query as a **string** inside of a Grafana dashboard panel's overall query, and reference the currently selected values of a multiple-value dashboard variable inside that query string. For example, my panel query included something like this:

```
DO $$
BEGIN
PERFORM some_postgresql_function('
   SELECT time, serial, parameter FROM some_table 
   WHERE time BETWEEN ''${__from:date}'' AND ''${__to:date}''
   AND serial IN (${grafana_multi_variable})
   GROUP BY serial'
);
END;
$$
```

Because the query is constructed as a SQL string, it's wrapped inside single quotes, which means any single quotes appearing inside the query need to be escaped, i.e. be **2** single quotes. So I'd need a way for the `${grafana_multi_variable}` to be expanded as something like

`''selected_val1'',''selected_val3'',''selected_val8''`

As far as I can tell from the [Advanced variable format options documentation](https://grafana.com/docs/grafana/latest/variables/advanced-variable-format-options/), there is no current option that can achieve this. 

Thus this feature request for such a variable formatting option. Perhaps it could be named `doublesinglequote` , i.e.

`${grafana_multi_variable:doublesinglequote}`

Incidentally, it would be nice if the advanced format options could be intelligently applied to the Grafana built-ins, particularly `$__timeFilter()` so that something like `$__timeFilter:doublesinglequote(time)` would spit out the `time BETWEEN ''…'' AND ''…''` form in the above query string, but I recognize that's a much more complicated change and it can be worked around as I have done above.

**Comment by gelicia** [comment]
Hello, as you may have heard, we are [transitioning away from using discussions to discuss feature requests](https://github.com/grafana/grafana/discussions/73424). Due to the age and number of responses to this discussion, we are deciding to close it. If this is something you would like to see in Grafana, feel free to open an [issue](https://github.com/grafana/grafana/issues) so the discussion can continue. Thank you!
